src/main/java/gov/usgs/ngwmn/logic/WaterLevelStatistics.java

```diff
@@ -175,16 +140,19 @@ public JsonData calculate(Specifier spec, List<WLSample> samples) {
 	}
 	
 
-	protected void overallStats(List<WLSample> samples, List<WLSample> sortedByValue) {
-		overallStatistics.overallStats(samples, sortedByValue);
-		if (samples == null || samples.size() == 0) {
+	protected void overallStats(List<WLSample> samplesByDate, List<WLSample> sortedByValue) {
+		overallStatistics.overallStats(samplesByDate, sortedByValue);
+		if (samplesByDate == null || samplesByDate.size() == 0) {
 			builder.recordYears("0");
 			builder.sampleCount(0);
 			builder.collect();
 			return;
 		}
+		WLSample latest = samplesByDate.get(samplesByDate.size()-1);
+		// TODO this might not work for any type???
+		BigDecimal pctl = percentileOfValue(sortedByValue, latest, Value::valueOf);
 		
-		String latestPercentile = monthlyStats.percentileBasedOnMonthlyData(samples.get(samples.size()-1), samples);
```

Comment:
This method almost works and has the right idea but it cannot be used for the latest percentile because of how it manipulates the data set.